packages/sw-cache-expiration/src/lib/behavior.js

```diff
@@ -14,144 +14,257 @@
+   * @param {Response} input.newResponse The new value in the cache.
    */
-  get db() {
-    if (!this._db) {
-      return idb.open(idbName, idbVersion, (upgradeDB) => {
-        upgradeDB.createObjectStore(this.cacheName);
-      }).then((db) => {
-        this._db = db;
-      });
-    }
+  cacheDidUpdate({cacheName, newResponse} = {}) {
+    assert.isType({cacheName}, 'string');
+    assert.isValue({cacheName}, this.cacheName);
+    assert.isInstance({newResponse}, Response);
 
-    return Promise.resolve(this._db);
+    const now = Date.now();
+    this.updateTimestamp({now, url: newResponse.url}).then(() => {
+      this.expireEntries({now});
+    });
   }
 
   /**
-   * TODO updateTimestamp description
-   * @private
-   * @param {string} url
-   * @param {Number} now Defaults to current date.
-   * @return {Promise} Resolves once updated.
+   * Updates the timestamp stored in IndexedDB for `url` to be equal to `now`.
+   *
+   * @param {Object} input The input object to this function.
+   * @param {string} input.url
+   * @param {Number} [input.now] A timestamp. Defaults to the current time.
    */
-  updateTimestamp(url, now) {
+  async updateTimestamp({url, now}) {
+    assert.isType({url}, 'string');
+
     if (typeof now === 'undefined') {
       now = Date.now();
     }
 
-    return this.db.then((db) => {
-      const tx = db.transaction(this.cacheName, 'readwrite');
-      tx.objectStore(this.cacheName).put(now, url);
-      return tx.complete;
+    const db = await this.getDB();
+    const tx = db.transaction(this.cacheName, 'readwrite');
+    tx.objectStore(this.cacheName).put({
+      [timestampProperty]: now,
+      [urlProperty]: url,
     });
+
+    await tx.complete;
   }
 
   /**
-   * TODO expireEntries description
-   * @param {Number} now Defaults to the current time
-   * @return {Promise} List of removed entries.
+   * Expires entries, both based on the the maximum age and the maximum number
+   * of entries, depending on how this instance is configured.
+   *
+   * @param {Object} input The input object to this function.
+   * @param {Number} [input.now] A timestamp. Defaults to the current time.
+   * @return {Array<string>} A list of the URLs that were expired.
    */
-  expireEntries(now) {
+  async expireEntries({now} = {}) {
     if (typeof now === 'undefined') {
       now = Date.now();
     }
 
-    const promises = [];
-    promises.push(
-      this.maxAgeSeconds ? this._expireOldEntries(now) : Promise.resolve([]));
-    promises.push(
-      this.maxEntries ? this._expireExtraEntries() : Promise.resolve([]));
+    // First, expire old entries, if maxAgeSeconds is set.
+    const oldEntries = this.maxAgeSeconds ?
+      await this.findOldEntries({now}) :
+      [];
+
+    // Once that's done, check for the maximum size.
+    const extraEntries = this.maxEntries ?
+      await this.findExtraEntries() :
+      [];
+
+    const urls = [...new Set(oldEntries.concat(extraEntries))];
```

Comment:
What is this code doing?

1. Concat the two arrays into a new array.
2. Make a Set from the new array.
3. Spread the set into an array.

Are the entries not just lists of URLs? If they are, can't step 2 and 3 just be skipped?